In [64]:
import numpy as np
import pandas as pd

In [65]:
df_events = pd.read_csv('G:/Rahul Data/GCD Related/CDFCapstone/Dataset/events_data.csv')
df_events.head()

event_id     device_id            timestamp  longitude   latitude  \
0   2765368  2.973348e+18  2016-05-07 22:52:05  77.225676  28.730140   
1   2955066  4.734221e+18  2016-05-01 20:44:16  88.388361  22.660325   
2    605968 -3.264500e+18  2016-05-02 14:23:04  77.256809  28.757906   
3    448114  5.731369e+18  2016-05-03 13:21:16  80.343613  13.153332   
4    665740  3.388880e+17  2016-05-06 03:51:05  85.997745  23.842609   

       city       state  
0     Delhi       Delhi  
1  Calcutta  WestBengal  
2     Delhi       Delhi  
3   Chennai   TamilNadu  
4    Bokaro   Jharkhand

In [66]:
df_events.shape

(3252950, 7)

In [67]:
df_events_states = df_events[(df_events['state'] == 'Bihar') |(df_events['state'] == 'Gujarat') | (df_events['state'] == 'Punjab') | (df_events['state'] == 'Kerala') |(df_events['state'] == 'WestBengal') ]

In [68]:
df_events_states.head()

event_id     device_id            timestamp  longitude   latitude  \
1    2955066  4.734221e+18  2016-05-01 20:44:16  88.388361  22.660325   
28    769546 -1.817502e+18  2016-05-01 14:07:23  88.371813  22.662852   
30   1750603 -5.598137e+18  2016-05-05 15:47:03  70.212684  23.118370   
31   3085968 -3.808297e+18  2016-05-07 01:25:47  75.513019  11.812367   
45   1447642 -8.778263e+18  2016-05-05 09:28:24  88.445866  22.610965   

          city       state  
1     Calcutta  WestBengal  
28    Calcutta  WestBengal  
30  Gandhidham     Gujarat  
31  Thalassery      Kerala  
45    Calcutta  WestBengal

In [69]:
df_events_states.shape

(323857, 7)

In [70]:
pip install mysql-connector-python

You should consider upgrading via the 'C:\Users\Snehal\anaconda3\python.exe -m pip install --upgrade pip' command.


In [71]:
import mysql.connector

In [72]:
connection = mysql.connector.connect(host="cpanel.insaid.co", database = 'Capstone1',user="student", passwd="student")
df_gender_age = pd.read_sql('''Select * from gender_age_train;''',connection)
df_brand_model = pd.read_sql('''Select * from phone_brand_device_model;''',connection)
connection.close()

In [73]:
df_gender_age.head()

device_id gender  age   group
0 -8076087639492063270      M   35  M32-38
1 -2897161552818060146      M   35  M32-38
2 -8260683887967679142      M   35  M32-38
3 -4938849341048082022      M   30  M29-31
4   245133531816851882      M   30  M29-31

In [74]:
df_brand_model.head()

device_id phone_brand device_model
0  1877775838486905855        vivo          Y13
1 -3766087376657242966          小米         V183
2 -6238937574958215831        OPPO          R7s
3  8973197758510677470          三星        A368t
4 -2015528097870762664          小米      红米Note2

In [75]:
df_gender_age.shape

(74645, 4)

In [76]:
df_brand_model.shape

(87726, 3)

In [77]:
df_gender_age[df_gender_age['device_id'] == 245133531816851882]

device_id gender  age   group
4  245133531816851882      M   30  M29-31

In [78]:
df_brand_model[df_brand_model['device_id'] == 245133531816851882]

device_id phone_brand device_model
13110  245133531816851882       SUGAR         时尚手机

In [79]:
df_gender_age_model = pd.merge(df_gender_age,df_brand_model,'inner',on='device_id')

In [80]:
df_gender_age_model.head()

device_id gender  age   group phone_brand device_model
0 -8076087639492063270      M   35  M32-38          小米         MI 2
1 -2897161552818060146      M   35  M32-38          小米         MI 2
2 -8260683887967679142      M   35  M32-38          小米         MI 2
3 -4938849341048082022      M   30  M29-31          三星    Galaxy S4
4   245133531816851882      M   30  M29-31       SUGAR         时尚手机

In [81]:
df_gender_age_model.shape

(74617, 6)

In [82]:
df_final = pd.merge(df_events_states,df_gender_age_model,on='device_id',how='left')

In [83]:
df_final.head()

event_id     device_id            timestamp  longitude   latitude  \
0   2955066  4.734221e+18  2016-05-01 20:44:16  88.388361  22.660325   
1    769546 -1.817502e+18  2016-05-01 14:07:23  88.371813  22.662852   
2   1750603 -5.598137e+18  2016-05-05 15:47:03  70.212684  23.118370   
3   3085968 -3.808297e+18  2016-05-07 01:25:47  75.513019  11.812367   
4   1447642 -8.778263e+18  2016-05-05 09:28:24  88.445866  22.610965   

         city       state gender  age group phone_brand device_model  
0    Calcutta  WestBengal    NaN  NaN   NaN         NaN          NaN  
1    Calcutta  WestBengal    NaN  NaN   NaN         NaN          NaN  
2  Gandhidham     Gujarat    NaN  NaN   NaN         NaN          NaN  
3  Thalassery      Kerala    NaN  NaN   NaN         NaN          NaN  
4    Calcutta  WestBengal    NaN  NaN   NaN         NaN          NaN

In [84]:
df_final.shape

(323857, 12)

In [85]:
import sweetviz as sv

In [86]:
data_analysis = sv.analyze(df_final)
data_analysis.show_html('analysis.html',open_browser=True)

C:\Users\Snehal\anaconda3\lib\site-packages\matplotlib\backends\backend_agg.py:238: RuntimeWarning: Glyph 21326 missing from current font.
  font.set_text(s, 0.0, flags=flags)
C:\Users\Snehal\anaconda3\lib\site-packages\matplotlib\backends\backend_agg.py:238: RuntimeWarning: Glyph 20026 missing from current font.
  font.set_text(s, 0.0, flags=flags)
C:\Users\Snehal\anaconda3\lib\site-packages\matplotlib\backends\backend_agg.py:238: RuntimeWarning: Glyph 19977 missing from current font.
  font.set_text(s, 0.0, flags=flags)
C:\Users\Snehal\anaconda3\lib\site-packages\matplotlib\backends\backend_agg.py:238: RuntimeWarning: Glyph 26143 missing from current font.
  font.set_text(s, 0.0, flags=flags)
C:\Users\Snehal\anaconda3\lib\site-packages\matplotlib\backends\backend_agg.py:238: RuntimeWarning: Glyph 23567 missing from current font.
  font.set_text(s, 0.0, flags=flags)
C:\Users\Snehal\anaconda3\lib\site-packages\matplotlib\backends\backend_agg.py:238: RuntimeWarning: Glyph 31859 missing 


Report analysis.html was generated! NOTEBOOK/COLAB USERS: the web browser MAY not pop up, regardless, the report IS saved in your notebook/colab files.


In [87]:
df_final.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 323857 entries, 0 to 323856
Data columns (total 12 columns):
 #   Column        Non-Null Count   Dtype  
---  ------        --------------   -----  
 0   event_id      323857 non-null  int64  
 1   device_id     323809 non-null  float64
 2   timestamp     323857 non-null  object 
 3   longitude     323815 non-null  float64
 4   latitude      323815 non-null  float64
 5   city          323857 non-null  object 
 6   state         323857 non-null  object 
 7   gender        2069 non-null    object 
 8   age           2069 non-null    float64
 9   group         2069 non-null    object 
 10  phone_brand   2069 non-null    object 
 11  device_model  2069 non-null    object 
dtypes: float64(4), int64(1), object(7)
memory usage: 32.1+ MB


Observations : 

1) There are three datasets, with 2 define the make of the mobile device and demographics of the user e.g. age, gender etc. The third dataset contain events information that are taking place from different devices.
2) Dataset contains 7 Object Features, 4 Flot Features and 1 Integer feature.
2) We segregated data of 5 states i.e. WestBengal, Gujarat, Punjab, Kerala and Bihar
3) Data of 3 lakh (0.3 million) records if available from 5 states and 
4) Data is collected for the year 2016
5) Data is collated from 250 different cities
6) Close to 6@% users are Male and approximately 38% users are female users
7) More than 95% of the users are less than or eual to 45 years of age whereas 75% of users are in the age group of 25 to 35 years old.
8) Users have used 13 different brands and 64 different device models are used.
9) 61% of the users are from WestBengal and 7% are from Kerala. 38% of the users are from Calcutta.  